cmd/gardener-metrics-exporter/app.go

```
@@ -122,30 +124,45 @@ func run(o *options, closeCh chan os.Signal) error {
 	return nil
 }
 
+func newConfigFromBytes(kubeconfig string) (*restclient.Config, error) {
+	kubecf, err := ioutil.ReadFile(kubeconfig)
+	if err != nil {
+		log.Errorf("Could not read config")
+	}
+	configObj, err := clientcmd.Load(kubecf)
+	if err != nil {
+		log.Errorf("Could not load config")
+	}
+	clientConfig := clientcmd.NewDefaultClientConfig(*configObj, &clientcmd.ConfigOverrides{})
+	client, err := clientConfig.ClientConfig()
+	if err != nil {
+		return nil, nil
```

Here we need to return the error `return nil, err`